File: tests/fixtures/ecosystems/csharp_comprehensive/Enums/Status.cs
using System;

namespace Comprehensive.Enums
{
    public enum Status
    {
        Active,
        Inactive,
        Pending,
        Deleted
    }

    [Flags]
    public enum Permissions
    {
        None = 0,
        Read = 1,
        Write = 2,
        Execute = 4,
        All = Read | Write | Execute
    }
}

File: tests/fixtures/ecosystems/java_comprehensive/enums/Status.java
package comprehensive.enums;

/**
 * Status enum with abstract method.
 */
public enum Status {
    ACTIVE {
        @Override
        public boolean isTerminal() {
            return false;
        }
    },
    INACTIVE {
        @Override
        public boolean isTerminal() {
            return false;
        }
    },
    DELETED {
        @Override
        public boolean isTerminal() {
            return true;
        }
    };

    public abstract boolean isTerminal();
}
